### PR TITLE
INTYGFV-15175: Implementation of getJsonFromUtlatande in the ModuleFacadeApi.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

**Comment by ChrilleSigerhed** [comment]
Seems like Ive made a mistake and included changes from another PR:  https://github.com/sklintyg/common/pull/696

Maybe review them before checking this one out.

File: support/src/main/java/se/inera/intyg/common/support/modules/support/facade/ModuleFacadeApi.java [hunk 1]
```diff
@@ -33,4 +34,6 @@ public interface ModuleFacadeApi {
     CertificateTextProvider getTextProvider(String certificateType, String certificateTypeVersion);
 
     CertificateMessagesProvider getMessagesProvider();
+
+    String getUtlatandeToInternalModelResponse(Utlatande utlatande) throws ModuleException;
```
Comment (mww-sopra-steria) [inline support/src/main/java/se/inera/intyg/common/support/modules/support/facade/ModuleFacadeApi.java:38]: Suggest renaming to getJsonFromUtlatande(Utlåtande utlåtande) to follow the naming convention.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v3/rest/TsDiabetesModuleApiV3.java [hunk 2]
```diff
@@ -211,4 +217,22 @@ public Certificate getCertificateFromJson(String certificateAsJson,
     public String getJsonFromCertificate(Certificate certificate, String certificateAsJson) throws ModuleException, IOException {
         throw new UnsupportedOperationException();
     }
+
+    @Override
+    public String getUtlatandeToInternalModelResponse(Utlatande utlatande) throws ModuleException {
+        if (utlatande instanceof TsDiabetesUtlatandeV3) {
+            return toInteralModelResponse((TsDiabetesUtlatandeV3) utlatande);
+        }
+        throw new IllegalArgumentException();
```
Comment (mww-sopra-steria) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v3/rest/TsDiabetesModuleApiV3.java:226]: Include a message with the class of the utlatande. Would be great to see what class the argument had in order to track down the culprit.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v3/rest/TsDiabetesModuleApiV3.java [hunk 3]
```diff
@@ -211,4 +217,22 @@ public Certificate getCertificateFromJson(String certificateAsJson,
...
+    }
+
+    private String toInteralModelResponse(TsDiabetesUtlatandeV3 internalModel) throws ModuleException {
```
Comment (mww-sopra-steria) [inline ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v3/rest/TsDiabetesModuleApiV3.java:229]: No need to implement it again. An implementation already exists in the TsParentModuleApi

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Great! Let's go with your solution. A few comments thought to look at first.

And a general comment. Implement the method in each of the certificate type/version (instead of throwing NotSupported-exception). All certificate types has an implementation of toInternalModelResponse available in its parent.

image

File: fk7263/src/main/java/se/inera/intyg/common/fk7263/rest/Fk7263ModuleApi.java [hunk 4]
```diff
@@ -713,4 +713,11 @@ public CertificateMessagesProvider getMessagesProvider() {
         return DefaultCertificateMessagesProvider.create(validationMessages);
     }
 
+    @Override
+    public String getUtlatandeToInternalModelResponse(Utlatande utlatande) throws ModuleException {
+        if (utlatande instanceof Fk7263Utlatande) {
+            return toInteralModelResponse((Fk7263Utlatande) utlatande);
```
Comment (mww-sopra-steria) [inline fk7263/src/main/java/se/inera/intyg/common/fk7263/rest/Fk7263ModuleApi.java:719]: Spelling error... maybe can fix now, even if it seems to exist from before.... toInteralModelResponse -> toInternalModelResponse.

File: support/src/main/java/se/inera/intyg/common/support/modules/support/facade/ModuleFacadeApi.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply support/src/main/java/se/inera/intyg/common/support/modules/support/facade/ModuleFacadeApi.java]: Done.

File: ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v3/rest/TsDiabetesModuleApiV3.java (hunk 3 above)
Comment (ChrilleSigerhed) [reply ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v3/rest/TsDiabetesModuleApiV3.java]: Done.

File: fk7263/src/main/java/se/inera/intyg/common/fk7263/rest/Fk7263ModuleApi.java (hunk 4 above)
Comment (ChrilleSigerhed) [reply fk7263/src/main/java/se/inera/intyg/common/fk7263/rest/Fk7263ModuleApi.java]: Done.

**Review by mikeyOne — APPROVED** [review]

**Review by mww-sopra-steria — APPROVED** [review]
Looks good!

**Pull request merged**